grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -65,4 +68,28 @@ class CycleCountApiController {
 
         render([data: cycleCount] as JSON)
     }
+
+    def updateCycleCountItem(CycleCountUpdateItemCommand command) {
+        if (command.hasErrors()) {
+            throw new ValidationException("Invalid cycle count item", command.errors)
+        }
+        CycleCountItemDto cycleCountItem = cycleCountService.updateCycleCountItem(command)
+
+        render([data: cycleCountItem] as JSON)
+    }
+
+    def deleteCycleCountItem() {
+        cycleCountService.deleteCycleCountItem(params.cycleCountItemId)
+
+        render(status: 204)
+    }
+
+    def createCustomCycleCountItem(CycleCountCustomItemCommand command) {
+        if (command.hasErrors()) {
+            throw new ValidationException("Invalid cycle count item", command.errors)
+        }
+        CycleCountItemDto cycleCountItem = cycleCountService.createCustomCycleCountItem(command)
```

Comment:
Could we save the conversion toDto until the render? I think I want the service to be returning CycleCount and CycleCountItem. Why do we even need the DTO at this point?

Reply:
In one of the earliest API tickets I explained that in one of the tech huddles - the point was that back then it was easier for me to work on DTOs even on service levels to shorten the data as much as I could, so that the flow was more understandable for me, especially that back then we had to build the cycle count dto via an item, hence I wanted to avoid having such logic in the marshallers, and it just kept going like this throughout further APIs.
I can change that here, but we would somehow lose the "convention" we have throughout this API.